packages/editor-ui/src/stores/settings.store.test.ts

```diff
@@ -0,0 +1,257 @@
+import type { FrontendSettings } from '@n8n/api-types';
+import { useSettingsStore } from './settings.store';
+import { createPinia, setActivePinia } from 'pinia';
+
+const { getSettings } = vi.hoisted(() => ({
+	getSettings: vi.fn(),
+}));
+
+const { sessionStarted } = vi.hoisted(() => ({
+	sessionStarted: vi.fn(),
+}));
+
+vi.mock('@/api/settings', () => ({
+	getSettings,
+}));
+
+vi.mock('@/api/events', () => ({
+	sessionStarted,
+}));
+
+vi.mock('@/stores/root.store', () => ({
+	useRootStore: vi.fn(() => ({
+		restApiContext: {},
+		setVersionCli: vi.fn(),
+	})),
+}));
+
+vi.mock('@/stores/root.store', () => ({
+	useRootStore: vi.fn(() => ({
+		setUrlBaseWebhook: vi.fn(),
+		setUrlBaseEditor: vi.fn(),
+		setEndpointForm: vi.fn(),
+		setEndpointFormTest: vi.fn(),
+		setEndpointFormWaiting: vi.fn(),
+		setEndpointWebhook: vi.fn(),
+		setEndpointWebhookTest: vi.fn(),
+		setEndpointWebhookWaiting: vi.fn(),
+		setTimezone: vi.fn(),
+		setExecutionTimeout: vi.fn(),
+		setMaxExecutionTimeout: vi.fn(),
+		setInstanceId: vi.fn(),
+		setOauthCallbackUrls: vi.fn(),
+		setN8nMetadata: vi.fn(),
+		setDefaultLocale: vi.fn(),
+		setBinaryDataMode: vi.fn(),
+		setVersionCli: vi.fn(),
+	})),
+}));
+
+vi.mock('@/stores/versions.store', () => ({
+	useVersionsStore: vi.fn(() => ({
+		setVersionNotificationSettings: vi.fn(),
+	})),
+}));
+
+const mockSettings: FrontendSettings = {
```

Comment:
Is there a better way to mock this huge object?

Reply:
pushed a commit to address this.

Reply:
Thank you!